subprojects/core/src/test/groovy/org/gradle/internal/cleanup/DefaultBuildOutputDeleterTest.groovy

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.cleanup
+
+import org.gradle.api.logging.Logger
+import org.gradle.test.fixtures.file.TestNameTestDirectoryProvider
+import org.gradle.util.GFileUtils
+import org.gradle.util.Requires
+import org.gradle.util.TestPrecondition
+import org.junit.Rule
+import spock.lang.Specification
+
+class DefaultBuildOutputDeleterTest extends Specification {
+
+    @Rule
+    final TestNameTestDirectoryProvider temporaryFolder = new TestNameTestDirectoryProvider()
+
+    def logger = Mock(Logger)
+    DefaultBuildOutputDeleter buildOutputDeleter = new DefaultBuildOutputDeleter()
+
+    def setup() {
+        buildOutputDeleter.logger = logger
```

Comment:
Maybe another/better way to do this (instead of relying on the logger sensing) is to make deleteOutput protected and override it here like:
```
def deletedFiles = []
def buildOutputDeleter = new DefaultBuildOutputDeleter() {
   void deleteOutput(File file) {
      deletedFiles << file
   }
}
```
Then you can assert that the correct things were deleted by inspecting `deletedFiles`.